core/app/models/spree/variant.rb

```diff
@@ -214,12 +222,40 @@ def option_value(opt_name)
       option_values.detect { |o| o.option_type.name == opt_name }.try(:presentation)
     end
 
+    # Returns an instance of the globally configured variant pricer class for this variant.
+    # It's cached so we don't create too many objects.
+    #
+    # @return [Spree::Variant::Pricer] The default pricer class
+    def pricer
+      @pricer ||= Spree::Config.variant_pricer_class.new(self)
+    end
+
+    # Chooses an appropriate price for the given pricing options
+    #
+    # @see Spree::Variant::Pricer#price_for
+    # @param [Spree::Config.pricing_options_class] An instance of pricing options
+    # @return [Spree::Money] The chosen price as a Money object
+    delegate :price_for, to: :pricer
+
+    # Returns the difference in price from the master variant
+    def price_difference_from_master(pricing_options = Spree::Config.default_pricing_options)
+      master_price = product.master.price_for(pricing_options)
+      variant_price = price_for(pricing_options)
+      return unless master_price && variant_price
+      variant_price - master_price
+    end
+
+    def price_same_as_master?(pricing_options = Spree::Config.default_pricing_options)
+      diff = price_difference_from_master(pricing_options)
+      diff && diff.zero?
+    end
+
```

Comment:
Once a user has, for example, a different tax address, these methods will need that user's or session's pricing options passed in here. I think that passing the argument is necessary (otherwise the user would get the price difference in the wrong currency, which would be wrong). I am, however, thinking it might be a good idea not to have a default argument and error out when these methods are called without arguments.